### ISSUE TITLE
Increase supported externals to include "in" "startswith" and "endswith"

### ISSUE THREAD

**Issue by tlansec** [VirusTotal/yara#1250]
Hello,

At the moment the following operators are supported for externals:

```
==
<
>
contains
matches
```
I think it would be useful to have an "in" operator in addition to these so that you can do:

```
my_custom_external in ['1', '2', '3']
```

This would be equivalent to:

```
my_custom_external == "1" or 
my_custom_external == "2" or 
my_custom_external == "3"
```

But may make for more readable rules in some cases.

Thanks,
Tom

**Comment by tlansec** [comment]
As an addendum to this, it would be useful to have:

```
"startswith"
"endswith" 
```

as string operators, that way when treating a filename as an external, for example i can use:

`filename endswith ".exe"`

rather than having to write:

`filename matches /\.exe$/`

Cheers,
Tom

**Comment by tlansec** [comment]
Are there any further thoughts on these suggestions?

I would imagine that there is greater incentive to add these now that the VT module is in play..

**Comment by plusvic** [comment]
Sounds like a very good idea. This is a low-hanging fruit, would you volunteer to implement it yourself?

**Comment by tlansec** [comment]
I have never written any C so it might be tough for me.

If I find time I might try, but i'll be starting from "Hello World". 

Cheers,
Tom

**Comment by plusvic** [comment]
OK, no problem then :)

**Comment by tlansec** [comment]
It looks like this is mostly (!? possibly completeld) implemented:

https://github.com/VirusTotal/yara/commit/41a32691d453bb4652f93d2e41bb5af1915cdda8

Linking here for documentation purposes and so you can close the issue more easily when appropriate.

**Comment by plusvic** [comment]
Implemented in #1357
